### ISSUE TITLE
Question: execute a Non-RootCmd directly?

### ISSUE THREAD

**Issue by Govinda-Fichtner** [spf13/cobra#237]
Is there a way to execute a Non-RootCmd directly?

```
# find a command
command, _, err := RootCmd.Find([]string{"set_hostname"})
# this executes the RootCmd always
# but I'd like to execute the current Cmd
err := command.Execute()
```

How could that be achieved?

**Comment by hypriot** [comment]
Well, I found another way to accomplish what wanted. So this issue can be closed.

**Comment by tj** [comment]
curious about this as well, seems that Execute etc are coupled with `os`, and no obvious decoupled alternative from what I can see

**Comment by frodopwns** [comment]
Something like

```
updateCmd.Run(RootCmd, []string{})
```

where updateCmd is a command added to the RootCmd?

**Comment by bogem** [comment]
Is the question answered? @Govinda-Fichtner

**Comment by Govinda-Fichtner** [comment]
@bogem I've been working around this ever since. As far as I can see there was not real answer to this question, but I don't mind if this issue gets closed.

**Comment by bogem** [comment]
> command, _, err := RootCmd.Find([]string{"set_hostname"})

Try to do this: 
```
command, _, err := RootCmd.Find([]string{RootCmd.Name(), "set_hostname"})
err := command.Execute()
```

**Comment by bogem** [comment]
Actually, it's a bug and not obvious behaviour. It happens because `Execute` func [executes only on root command](https://github.com/spf13/cobra/blob/master/command.go#L675).
The only solution I found is
```go
args := []string{"set_hostname"}
command, _, err := RootCmd.Find(args)
RootCmd.SetArgs(args)
err := command.Execute()
```

**Comment by sb10** [comment]
How do you also get it to react to the non-root-cmd's flags?

I have a RootCmd with no Run so it just prints some help text. I have a nested sub commands for which you can get help with `-h`.

I need to symlink my app and when my app is being called with a symlink of a certain name, I need it to behave as if `myapp subcmd subsubcmd -flags` was called.

With the above work-around and some detection in my main.go, `symlinkToMyApp -h` results in `myapp subcmd subsubcmd` being run, ie. ignoring the `-h`.

**Comment by johnSchnake** [comment]
The OP said the issue could be closed years ago so per #1600 I'm going to go ahead and close. If someone needs to reopen to discuss feel free and just reiterate the scope of your specific question.
